File: src/main/java/io/github/tanguygab/tabadditions/shared/features/chat/ChatFormat.java
package io.github.tanguygab.tabadditions.shared.features.chat;

import io.github.tanguygab.tabadditions.shared.features.advancedconditions.AdvancedConditions;
import lombok.AllArgsConstructor;
import lombok.Getter;
import me.neznamy.tab.shared.platform.TabPlayer;

@AllArgsConstructor
public class ChatFormat {

    @Getter private final String name;
    @Getter private final String displayName;
    private final AdvancedConditions condition;
    @Getter private final AdvancedConditions viewCondition;
    @Getter private final String channel;
    @Getter private final String text;

    public boolean isConditionMet(TabPlayer p) {
        return condition == null || condition.isMet(p);
    }

    public boolean hasNoViewCondition() {
        return viewCondition == null;
    }

    public boolean isViewConditionMet(TabPlayer sender, TabPlayer viewer) {
        return hasNoViewCondition() || viewCondition.isMet(viewer,sender);
    }
}

File: src/main/kotlin/io/github/tanguygab/tabadditions/shared/features/chat/ChatFormat.kt
package io.github.tanguygab.tabadditions.shared.features.chat

import me.neznamy.tab.shared.placeholders.conditions.Condition
import me.neznamy.tab.shared.platform.TabPlayer

open class ChatFormat(
    val name: String,
    val displayName: String,
    private val condition: Condition?,
    val viewCondition: Condition?,
    val channel: String,
    val text: String
) {
    fun isConditionMet(p: TabPlayer) = condition?.isMet(p) ?: true
    fun hasNoViewCondition() = viewCondition == null
    fun isViewConditionMet(sender: TabPlayer, viewer: TabPlayer) = hasNoViewCondition() || viewCondition!!.isMet(viewer, sender)
}
